pkg/entrypoint/admissioncontroller/podwhitelist.go

```diff
@@ -0,0 +1,145 @@
+	unpackers := map[string]func(runtime.Object) (core.PodSpec, metav1.ObjectMeta, string){
+		"Pod": func(o runtime.Object) (core.PodSpec, metav1.ObjectMeta, string) {
+			pod := o.(*core.Pod)
+			return pod.Spec, pod.ObjectMeta, pod.Namespace
+		},
+		"DaemonSet": func(o runtime.Object) (core.PodSpec, metav1.ObjectMeta, string) {
+			ds := o.(*extensions.DaemonSet)
+			return ds.Spec.Template.Spec, ds.Spec.Template.ObjectMeta, ds.Namespace
+		},
+		"ReplicaSet": func(o runtime.Object) (core.PodSpec, metav1.ObjectMeta, string) {
+			rs := o.(*extensions.ReplicaSet)
+			return rs.Spec.Template.Spec, rs.Spec.Template.ObjectMeta, rs.Namespace
+		},
+		"StatefulSet": func(o runtime.Object) (core.PodSpec, metav1.ObjectMeta, string) {
+			ss := o.(*apps.StatefulSet)
+			return ss.Spec.Template.Spec, ss.Spec.Template.ObjectMeta, ss.Namespace
+		},
+		"Job": func(o runtime.Object) (core.PodSpec, metav1.ObjectMeta, string) {
+			job := o.(*batch.Job)
+			return job.Spec.Template.Spec, job.Spec.Template.ObjectMeta, job.Namespace
+		},
+		"CronJob": func(o runtime.Object) (core.PodSpec, metav1.ObjectMeta, string) {
+			cj := o.(*batch.CronJob)
+			return cj.Spec.JobTemplate.Spec.Template.Spec, cj.Spec.JobTemplate.Spec.Template.ObjectMeta, cj.Namespace
+		},
+		"DeploymentConfig": func(o runtime.Object) (core.PodSpec, metav1.ObjectMeta, string) {
+			dc := o.(*oapps.DeploymentConfig)
+			return dc.Spec.Template.Spec, dc.Spec.Template.ObjectMeta, dc.Namespace
+		},
+		"Deployment": func(o runtime.Object) (core.PodSpec, metav1.ObjectMeta, string) {
+			dp := o.(*extensions.Deployment)
+			return dp.Spec.Template.Spec, dp.Spec.Template.ObjectMeta, dp.Namespace
+		},
+	}
+	req, errcode := ac.getAdmissionReviewRequest(r)
+	if errcode != 0 {
+		http.Error(w, http.StatusText(errcode), errcode)
+		return
+	}
+	if req.UID == "" || req.Kind.Version == "" || req.Kind.Kind == "" {
+		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+		return
+	}
+	gvk := schema.GroupVersionKind{Group: req.Kind.Group, Version: req.Kind.Version, Kind: req.Kind.Kind}
+	o, gvkDecoded, err := codec.Decode(req.Object.Raw, &gvk, nil)
+	if err != nil {
+		log.Printf("Decode error:  %s", err)
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	unpackingFunc, found := unpackers[gvkDecoded.Kind]
+	if !found {
+		http.Error(w, http.StatusText(http.StatusNotImplemented), http.StatusNotImplemented)
+	}
+	spec, meta, namespace := unpackingFunc(o)
+
+	ac.checkPodSpec(spec, meta, namespace, w, req.UID)
+}
+
+//checkPodSpec checks if the Pod spec is either whitelisted or will match the restricted scc, then prepares an HTTP response
+// interface{} is used to allow core.Pod from both the Openshift and Kubernetes APIs
+func (ac *admissionController) checkPodSpec(podSpec core.PodSpec, oMeta metav1.ObjectMeta, namespace string, w http.ResponseWriter, uid types.UID) {
+	pod := new(core.Pod)
```

Comment:
this will boil away with the above changes to unpack()

Reply:
Done.